Minio/DataModel/Select/SelectResponseStream.cs

```diff
@@ -0,0 +1,239 @@
+            }
+            this._isProcessing = true;
+            this.payloadStream.Seek(0, SeekOrigin.Begin);
+            this.start();
+        }
+        
+        private void start()
+        {
+            int numBytesRead = 0;
+            int msgOffset = 0;
+            while (_isProcessing)
+            {   
+                var n =  ReadFromStream(prelude);
+                numBytesRead += n;
+                n =  ReadFromStream(preludeCRC);
+                numBytesRead += n;
+                var inputArray = new byte[prelude.Length + preludeCRC.Length + 4];
+                System.Buffer.BlockCopy(prelude,0,inputArray,0, prelude.Length);
+                System.Buffer.BlockCopy(preludeCRC,0,inputArray,prelude.Length, preludeCRC.Length);
+
+                // write real data to inputArray
+                Crc32Algorithm.ComputeAndWriteToEnd(inputArray); // last 4 bytes contains CRC
+                // transferring data or writing reading, and checking as final operation
+                if (!Crc32Algorithm.IsValidWithCrcAtEnd(inputArray))
+                {
+                    throw new ArgumentException("Prelude CRC Mismatch");
+                }
+                 var bytes = prelude.Take(4).ToArray();
+                 if (BitConverter.IsLittleEndian)
+                 {
+                    Array.Reverse(bytes); 
+                 }
+                int totalLength = BitConverter.ToInt32(bytes, 0);
+                bytes = prelude.Skip(4).Take(4).ToArray();
+                if (BitConverter.IsLittleEndian)
+                {
+                    Array.Reverse(bytes); 
+                }
+
+                int headerLength = BitConverter.ToInt32(bytes, 0);
+                int payloadLength = totalLength - headerLength - 16;
+
+                var headers = new byte[headerLength];
+                var payload  = new byte[payloadLength];
+                int num = ReadFromStream(headers);
+                if (num != headerLength)
+                {
+                    throw new IOException("insufficient data");
+
+                }
+                num = ReadFromStream(payload);
+                if (num != payloadLength)
+                {
+                    throw new IOException("insufficient data");
+                }
+
+                numBytesRead += num;
+                num = ReadFromStream(messageCRC);
+                // now verify message CRC
+                inputArray = new byte[totalLength + 4];
+                System.Buffer.BlockCopy(prelude,0,inputArray,0, prelude.Length);
+                System.Buffer.BlockCopy(preludeCRC,0,inputArray,prelude.Length, preludeCRC.Length);
+                System.Buffer.BlockCopy(headers,0,inputArray,prelude.Length+ preludeCRC.Length, headerLength);
+                System.Buffer.BlockCopy(payload,0,inputArray,prelude.Length+ preludeCRC.Length+headerLength, payloadLength);
+                System.Buffer.BlockCopy(messageCRC,0,inputArray, totalLength - 4,messageCRC.Length);
```

Comment:
We should exclude messageCRC for the crc calculation here as well

Reply:
fixed